rusk-recovery/src/keys.rs

```diff
@@ -32,12 +30,23 @@ static PUB_PARAMS: Lazy<PublicParameters> = Lazy::new(|| {
 
         _ => {
             warn!("{} new CRS due to cache miss", theme.warn("Building"));
-            let mut rng = StdRng::seed_from_u64(0xbeef);
 
-            let pp = PublicParameters::setup(1 << 17, &mut rng)
+            let (tx, rx) = mpsc::channel();
+
+            thread::spawn(move || {
+                let pp_bytes =
+                    fetch_pp().expect("PublicParameters download failed.");
+                tx.send(pp_bytes).unwrap();
+            })
+            .join()
+            .expect("PublicParameters download thread panicked");
```

Comment:
why are you spawning a dedicated thread? Is it due to the lazy initialization?

Reply:
it didn't allow me to execute it due to "runtime from within runtime"